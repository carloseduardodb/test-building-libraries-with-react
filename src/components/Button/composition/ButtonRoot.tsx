import React, { ButtonHTMLAttributes, ReactNode } from 'react';

interface ButtonProps extends ButtonHTMLAttributes<HTMLButtonElement> {
  children?: ReactNode;
  styleType?: 'xs' | 'md' | 'lg';
}

export const ButtonRoot = ({
  children,
  styleType = 'md',
  ...props
}: ButtonProps) => {
  let buttonClass = '';

  switch (styleType) {
    case 'xs':
      buttonClass =
        'max-w-48 bg-blue-400 hover:bg-blue-600 text-white font-bold py-1 px-2 rounded';
      break;
    case 'md':
      buttonClass =
        'max-w-96 bg-blue-500 hover:bg-blue-700 text-white font-bold py-2 px-4 rounded';
      break;
    case 'lg':
      buttonClass =
        'max-w-120 bg-blue-600 hover:bg-blue-800 text-white font-bold py-3 px-6 rounded';
      break;
    default:
      break;
  }

  return (
    <button className={buttonClass} type="button" {...props}>
      {children}
    </button>
  );
};
