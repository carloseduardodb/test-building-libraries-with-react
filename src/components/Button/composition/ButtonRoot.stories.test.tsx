import renderer from 'react-test-renderer';
import React from 'react';
import { ButtonRoot } from './ButtonRoot';

describe('ButtonRoot', () => {
  it('test_default_style_type', () => {
    const tree = renderer.create(<ButtonRoot />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('test_custom_children', () => {
    const tree = renderer
      .create(<ButtonRoot>Custom Children</ButtonRoot>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('test_custom_style_type_xs', () => {
    const tree = renderer.create(<ButtonRoot styleType="xs" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('test_custom_style_type_lg', () => {
    const tree = renderer.create(<ButtonRoot styleType="lg" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('test_empty_children_prop', () => {
    const tree = renderer.create(<ButtonRoot> </ButtonRoot>).toJSON();
    expect(tree).toMatchSnapshot(' ');
  });
});
