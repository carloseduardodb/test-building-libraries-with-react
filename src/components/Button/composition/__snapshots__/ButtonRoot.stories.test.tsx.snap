// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonRoot test_all_props_undefined 1`] = `
<button
  className="max-w-96 bg-blue-500 hover:bg-blue-700 text-white font-bold py-2 px-4 rounded"
  type="button"
/>
`;

exports[`ButtonRoot test_custom_children 1`] = `
<button
  className="max-w-96 bg-blue-500 hover:bg-blue-700 text-white font-bold py-2 px-4 rounded"
  type="button"
>
  Custom Children
</button>
`;

exports[`ButtonRoot test_custom_style_type_lg 1`] = `
<button
  className="max-w-120 bg-blue-600 hover:bg-blue-800 text-white font-bold py-3 px-6 rounded"
  type="button"
/>
`;

exports[`ButtonRoot test_custom_style_type_xs 1`] = `
<button
  className="max-w-48 bg-blue-400 hover:bg-blue-600 text-white font-bold py-1 px-2 rounded"
  type="button"
/>
`;

exports[`ButtonRoot test_default_style_type 1`] = `
<button
  className="max-w-96 bg-blue-500 hover:bg-blue-700 text-white font-bold py-2 px-4 rounded"
  type="button"
/>
`;

exports[`ButtonRoot test_empty_children_prop 1`] = `
<button
  className="max-w-96 bg-blue-500 hover:bg-blue-700 text-white font-bold py-2 px-4 rounded"
  type="button"
/>
`;

exports[`ButtonRoot test_empty_children_prop:   1`] = `
<button
  className="max-w-96 bg-blue-500 hover:bg-blue-700 text-white font-bold py-2 px-4 rounded"
  type="button"
>
   
</button>
`;
